api/api.proto

```diff
@@ -2684,3 +2686,74 @@ message LocateResponse {
     // <containerid>: /var/www
     map<string, string> dockerids = 2;
 }
+
+message VolumePlacementSpec {
+  // Defines a list of rules as part of the placement spec
+  repeated VolumePlacementRule rules = 1;
+}
+
+message VolumePlacementRule {
+  // Defines the number of volume replicas affected by this rule. If not provided,
+  // rule would affect all replicas
+  // +optional
+  uint64 affectedReplicas = 1;
+  // Defines the affinity rule for the volume placement
+  VolumePlacementAffinity affinity = 2;
+}
+
+// VolumePlacementAffinity allows to specify affinity and anti-affinity rules based on
+// storage pools, nodes and volumes.
+message VolumePlacementAffinity {
+  // Defines the storage pool affinity for volume placement
+  AffinityRule poolAffinity = 1;
```

Comment:
This is an example, but all names should not use camel case. Instead they should use `_` and the Golang generator changes them to camel case.